django/forms/widgets.py

```diff
@@ -976,39 +976,42 @@ def get_context(self, name, value, attrs):
         if not self.is_required:
             year_choices.insert(0, self.year_none_value)
         year_name = self.year_field % name
-        date_context['year'] = self.select_widget(attrs, choices=year_choices).get_context(
+        year_widget = self.select_widget(attrs, choices=year_choices)
+        year_widget.is_required = self.is_required
+        date_context['year'] = year_widget.get_context(
             name=year_name,
             value=context['widget']['value']['year'],
             attrs={
                 **context['widget']['attrs'],
                 'id': 'id_%s' % year_name,
-                'placeholder': _('Year') if self.is_required else False,
```

Comment:
Do we not need logic to pass the `required` attribute here?

Reply:
This is handled by `BoundField`.

https://github.com/django/django/blob/f97a6123c07de5099fdf8b7d00ef7d20ed354e07/django/forms/boundfield.py#L224-L225